Mathlib/RingTheory/Radical.lean

```diff
@@ -104,4 +106,57 @@ theorem radical_pow_of_prime {a : M} (ha : Prime a) {n : ℕ} (hn : 0 < n) :
   rw [radical_pow a hn]
   exact radical_of_prime ha
 
+
+-- Theorems for commutative rings
+
+-- TODO: This holds for "nontrivial" monoids - do not need ring assumption.
+theorem radical_ne_zero (a : R) : radical a ≠ 0 := by
+  rw [radical, ← Finset.prod_val]
+  apply Multiset.prod_ne_zero
+  rw [primeFactors]
+  simp only [Multiset.toFinset_val, Multiset.mem_dedup]
+  exact zero_not_mem_normalizedFactors _
+
+/-- Coprime elements have disjoint prime factors (as multisets). -/
+theorem disjoint_normalizedFactors {a b : R} (hc : IsCoprime a b) :
+    (normalizedFactors a).Disjoint (normalizedFactors b) := by
+  intro x hxa hxb
+  have x_dvd_a := dvd_of_mem_normalizedFactors hxa
+  have x_dvd_b := dvd_of_mem_normalizedFactors hxb
+  have xp := prime_of_normalized_factor x hxa
+  exact xp.not_unit (hc.isUnit_of_dvd' x_dvd_a x_dvd_b)
+
+/-- Coprime elements have disjoint prime factors (as finsets). -/
+theorem disjoint_primeFactors {a b : R} (hc : IsCoprime a b) :
+    Disjoint (primeFactors a) (primeFactors b) :=
+  Multiset.disjoint_toFinset.mpr (disjoint_normalizedFactors hc)
+
+theorem hMul_primeFactors_disjUnion {a b : R} (ha : a ≠ 0) (hb : b ≠ 0)
+    (hc : IsCoprime a b) :
+    primeFactors (a * b) = (primeFactors a).disjUnion (primeFactors b) (disjoint_primeFactors hc)
+    := by
+  rw [Finset.disjUnion_eq_union]
+  simp_rw [primeFactors]
+  rw [normalizedFactors_mul ha hb, Multiset.toFinset_add]
+
+@[simp]
+theorem radical_neg_one : radical (-1 : R) = 1 :=
+  radical_unit_eq_one isUnit_one.neg
+
+/-- Radical is multiplicative for coprime elements. -/
+theorem radical_hMul {a b : R} (hc : IsCoprime a b) :
```

Comment:
Is there a reason to use `hMul` rather than `mul` in this name?

Reply:
Actually not, I think `mul` is better in this case. I just updated.